src/System.Security.Cryptography.Algorithms/tests/HashAlgorithmTest.cs

```diff
@@ -54,6 +55,154 @@ private void VerifyICryptoTransformStream(Stream input, string output)
         }
 #endif
 
+        protected void VerifyMultiBlock(string block1, string block2, string expectedHash, string emptyHash)
+        {
+#if netstandard17
+            byte[] block1_bytes = ByteUtils.AsciiBytes(block1);
+            byte[] block2_bytes = ByteUtils.AsciiBytes(block2);
+            byte[] expected_bytes = ByteUtils.HexToByteArray(expectedHash);
+            byte[] emptyHash_bytes = ByteUtils.HexToByteArray(emptyHash);
+
+            VerifyTransformOutput(block1_bytes, block2_bytes);
+            VerifyTransformComputeHashInteraction(block1_bytes, block2_bytes, expected_bytes, emptyHash_bytes);
+#endif
+        }
+
+#if netstandard17
+        private void VerifyTransformOutput(byte[] block1, byte[] block2)
+        {
+            byte[] output = new byte[block1.Length + block2.Length];
+
+            // Concat block1 and block2
+            byte[] expected = new byte[block1.Length + block2.Length];
+            Buffer.BlockCopy(block1, 0, expected, 0, block1.Length);
+            Buffer.BlockCopy(block2, 0, expected, block1.Length, block2.Length);
+
+            using (HashAlgorithm hash = Create())
+            {
+                int byteCount = hash.TransformBlock(block1, 0, block1.Length, output, 0);
+                hash.TransformBlock(block2, 0, block2.Length, output, byteCount);
+                Assert.Equal(expected, output);
+            }
+        }
+
+        private void VerifyTransformComputeHashInteraction(byte[] block1, byte[] block2, byte[] expected, byte[] expectedEmpty)
+        {
+            using (HashAlgorithm hash = Create())
+            {
+                // TransformBlock + ComputeHash
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                byte[] actual = hash.ComputeHash(block2, 0, block2.Length);
+                Assert.Equal(expected, actual);
+
+                // Verify ComputeHash cleared hash
+                actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
+                Assert.Equal(expectedEmpty, actual);
+
+                // Re-hash the same data
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                actual = hash.ComputeHash(block2, 0, block2.Length);
+                Assert.Equal(expected, actual);
+
+                // Verify TransformFinalBlock
+                hash.TransformBlock(block1, 0, block1.Length, null, 0);
+                hash.TransformFinalBlock(block2, 0, block2.Length);
+                Assert.Equal(expected, hash.Hash);
+                Assert.Equal(expected, hash.Hash); // .Hash doesn't clear hash
+
+                // Todo: this will also fail due to the ActiveIssue below in VerifyUseAfterTransformFinalBlock
+                // but commented out so the other tests can run
+                // actual = hash.ComputeHash(Array.Empty<byte>(), 0, 0);
```

Comment:
yes corefx=empty, netfx=tainted\same as last. On whether ComputeHash is necessary on netfx for the next call - yes it is (that's what the ActiveIssue \ VerifyUseAfterTransformFinalBlock method checks for)